### PR TITLE
Marcello oauth

### PR DESCRIPTION
**Pull request by marcellosachs**
@wlaurance @kriskelly made a calculatePercentageService so that a percentage is calculated within the finalizeService. Does everything here look ok? Thanks, Marcello

### CODE REVIEW

File: src/scripts/services/calculatePercentage.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports =
+angular.module("quill-grammar.services.calculatePercentage", [])
+.factory('calculatePercentageService', function () {
+  var calculatePercentage = function (conceptTagResults) {
+    var numCorrect, percentage;
+    if (conceptTagResults.length == 0) {
```
Comment (kriskelly) [inline src/scripts/services/calculatePercentage.js]: More idiomatic would be something like:

```
if (conceptTagResults.length) {
  // do score counting
} else {
 percentage = 0;
}
```

**Comment by kriskelly** [comment]
@marcellosachs You should run the linter and fix all the linting errors before this can be merged.

```
./node_modules/.bin/grunt lint
```

**Comment by marcellosachs** [comment]
@kriskelly k gotcha...

File: src/scripts/services/calculatePercentage.js (hunk 1 above)
Comment (marcellosachs) [inline src/scripts/services/calculatePercentage.js]: the issue isnt when the length is undefined, its when the length is 0 (in which case the percentage calculating will involve a 0 in the denominator)

Comment (marcellosachs) [inline src/scripts/services/calculatePercentage.js]: I spoke to soon - realize now that 0 is 'falsey'. Gotcha..

**Comment by marcellosachs** [comment]
@kriskelly ok took care of lint errors and switched to the idiomatic 'if' pattern. Anything else I should change?

**Comment by marcellosachs** [comment]
ok with last commit everything is lint-free...

**Pull request merged**